Gems/ROS2/Code/Platform/Common/Clang/ros2_static_clang.cmake

```diff
@@ -0,0 +1,11 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+
+set(LY_COMPILE_OPTIONS
```

Comment:
We should avoid adding exception dependency in non-editor target:
https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#c-exceptions

I believe this could be avoided completely by following some other of my comments

Reply:
> @danielemorra98 I have a hard time compiling this change against ROSConDemo:
> 
> https://gist.github.com/michalpelka/4f4cf5282aa8a098ca27bd1be9e3168f
> 
> It seems that we need to have exceptions `-fexceptions` enabled in-game targets.

This file was added in order to make ROSConDemo project compile. I did not use exceptions inside my code, instead they come from `rclcpp_action implementations` (see here: https://gist.github.com/michalpelka/4f4cf5282aa8a098ca27bd1be9e3168f#file-compilation-error-L10).